### ISSUE TITLE
[api-extractor] re-exported modules are not include in .d.ts rollup

### ISSUE THREAD

**Issue by gesiu** [Microsoft/web-build-tools#1148]
When an imported module is re-exported, the .d.ts rollup generated by dtsRollup does not include the re-exported module:

Sample code:
```
import { Foo } from './Foo';

import _Foo = Foo;

export module Bar {
    export import Foo = _Foo;
}
```

Resulting .d.ts:
```
export declare module Bar {
    export import Foo = _Foo;
}
```

The code in Foo.ts is lost. See [https://github.com/gesiu/temp] for a repro.

**Comment by octogonz** [comment]
Is there a workaround?  For example, does this work?

```ts
import { Foo as _Foo } from './Foo';

export module Bar {
    export import Foo = _Foo;
}
```

**Comment by gesiu** [comment]
That doesn't appear to work either. The resulting .d.ts now also drops the export:
```
export declare module Bar {
    import Foo = _Foo;
}
```

**Comment by deini** [comment]
We are experiencing this issue in one of our projects, just wanted to leave our use case here for some context.

We are using `@babel/preset-typescript` with `isolatedModules`. Our setup has limitations when re-exporting types. So we have to do something like:

```ts
import { BoxProps as _BoxProps } from './Box';

export { Box } from './Box';
export type BoxProps = _BoxProps;
```

When running `api-extractor` with this setup we hit this issue.

**Comment by octogonz** [comment]
@deini Your example does not seem to be using a `module`, so it may be a different issue.  The code that you pasted looks like it should work correctly.

Could you share a branch that repros your issue, and include an error message or incorrect output that shows the problem you are encountering?

**Comment by deini** [comment]
Good point, might be something different. I will try to create a small repro and open a new issue if I can reproduce it. Probably just a user error on my end.
